### PR TITLE
Fix #324: Submit answer upon keyboard enter 

### LINKED ISSUE #324: Fix submit the answer upon keyboard enter
There’s currently a crash that occurs in the exploration player when you press the IME soft keyboard return button. This seems to be a focus related crash, but it’s not clear why. The return button should submit an answer for each of the text based interactions.

### PR DESCRIPTION
**Pull request by veena14cs**
## Explanation
This PR submits the answer upon keyboard enter.

Note: Validation is not yet done on textinput interactions as per #376.

### CODE REVIEW

File: app/src/main/java/org/oppia/app/player/exploration/ExplorationActivityPresenter.kt [hunk 1]
```diff
@@ -9,11 +10,14 @@ import androidx.lifecycle.Transformations
 import org.oppia.app.R
 import org.oppia.app.activity.ActivityScope
 import org.oppia.app.model.Exploration
+import org.oppia.app.player.state.StateFragment
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/player/exploration/ExplorationActivityPresenter.kt]: Remove this unused import.

File: app/src/main/java/org/oppia/app/player/state/StateFragment.kt [hunk 2]
```diff
@@ -53,4 +53,7 @@ class StateFragment : InjectableFragment(), CellularDataInterface, InteractionAn
   }
 
   fun handlePlayAudio() = stateFragmentPresenter.handleAudioClick()
+
+  fun handleKeyboardAction() = stateFragmentPresenter.handleKeyboardAction()
+
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/player/state/StateFragment.kt]: Remove this extra blank line.

File: app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt [hunk 3]
```diff
@@ -4,6 +4,7 @@ import android.content.Context
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.view.inputmethod.EditorInfo
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt]: Remove this unused import

File: app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt [hunk 4]
```diff
@@ -45,6 +46,7 @@ import org.oppia.app.player.state.itemviewmodel.StateItemViewModel
 import org.oppia.app.player.state.itemviewmodel.StateNavigationButtonViewModel
 import org.oppia.app.player.state.itemviewmodel.StateNavigationButtonViewModel.ContinuationNavigationButtonType
 import org.oppia.app.player.state.itemviewmodel.TextInputViewModel
+import org.oppia.app.player.state.listener.StateKeyboardButtonListener
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt]: Ditto

**Review by rt4914 — APPROVED** [review]
LGTM, nit changes suggested.

File: app/src/main/java/org/oppia/app/player/exploration/ExplorationActivityPresenter.kt (hunk 1 above)
Comment (veena14cs) [reply app/src/main/java/org/oppia/app/player/exploration/ExplorationActivityPresenter.kt]: done.

File: app/src/main/java/org/oppia/app/player/state/StateFragment.kt (hunk 2 above)
Comment (veena14cs) [reply app/src/main/java/org/oppia/app/player/state/StateFragment.kt]: done.

File: app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt (hunk 3 above)
Comment (veena14cs) [reply app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt]: Removed.

File: app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt (hunk 4 above)
Comment (veena14cs) [reply app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt]: Done.

**Review by nikitamarysolomanpvt — APPROVED** [review]
LGTM

**Pull request merged**